databroker/core.py

```diff
@@ -26,6 +26,7 @@
 from .intake_xarray_core.base import DataSourceMixin
 from .intake_xarray_core.xarray_container import RemoteXarray
 from collections import deque
+from dask.base import normalize_token
```

Comment:
How hard is our dask dependency currently? I didn't think we had a hard dep, does intake?

Independent of if intake has a hard dependency, it is probably better to hide this import and the registration of the tokenize methods in a `try: .... except ImportError: ...` block.

Reply:
Intake has a [hard dependency on `dask[base]`](https://github.com/intake/intake/blob/34dd38089ccb805759508659b5e8475034df2f0c/requirements.txt#L4), which pulls in a few things but notable does *not* pull in heavy stuff like pandas.

Note that it is possible to load data as documents without _using_ dask, but databroker does now require that the user _install_ it.

Reply:
if we are going to directly import dask, we should also have a hard dependency on what we use (or bury the imports, I'm cool either way).

Reply:
My wording was poor. Intake has a hard dependency on `dask[base]` but _also_ databroker has a hard dependency on `dask[array,base]` already declared in current `master`.

Reply:
ah, fair enough.